# Import Libraries

In [138]:
import pandas as pd
import numpy as np
import re

In [139]:
chosun = pd.read_csv('chosun_news.csv')
print(chosun)
chosun.shape

        index                                                  0        date
0           0                      경제전문가·일반인 긴급 여론조사;“구조조정만이 살길”  2001-01-01
1           1                            ◇희망의 2001년… 다시 한번 뛰어보자;  2001-01-01
2           2                       클린턴이 임명한 테닛;부시, CIA국장 유임시킬듯;  2001-01-01
3           3  민주의원 이적 새해정국 파문;3명 자민련 입당… 교섭단체 가능 야 강력반발… 영수회...  2001-01-01
4           4              공적자금 투입 4개은 금융노조 전격 탈퇴;한빛·서울도 구조조정 동의  2001-01-01
...       ...                                                ...         ...
929118     88  [강헌의 히스토리 인 팝스] (138) Michelle Obama 'This is ...  2022-11-21
929119     89               [社說] 공공연히 ＂윤석열 퇴진＂ 민주당, 대선 불복하겠다는 건가  2022-11-21
929120     90               [社說] '韓 전술핵' 논의하는 것 자체가 북·중에 메시지 될 것  2022-11-21
929121     91                [社說] '복심' 정진상 구속, 李대표 턱밑까지 온 대장동 수사  2022-11-21
929122     92                      [전면광고] 세상을 바꾸는 K-BAB - KB금융그룹  2022-11-21

[929123 rows x 3 columns]


(929123, 3)

In [140]:
chosun['0'].nunique() # 중복 데이터 삭제 필요 (929123 - 865633 = 63490 개 중복 데이터 존재 확인)

865633

In [141]:
data = chosun.drop([chosun.columns[0]], axis = 1) # index 열 삭제
data

0        date
0                           경제전문가·일반인 긴급 여론조사;“구조조정만이 살길”  2001-01-01
1                                 ◇희망의 2001년… 다시 한번 뛰어보자;  2001-01-01
2                            클린턴이 임명한 테닛;부시, CIA국장 유임시킬듯;  2001-01-01
3       민주의원 이적 새해정국 파문;3명 자민련 입당… 교섭단체 가능 야 강력반발… 영수회...  2001-01-01
4                   공적자금 투입 4개은 금융노조 전격 탈퇴;한빛·서울도 구조조정 동의  2001-01-01
...                                                   ...         ...
929118  [강헌의 히스토리 인 팝스] (138) Michelle Obama 'This is ...  2022-11-21
929119               [社說] 공공연히 ＂윤석열 퇴진＂ 민주당, 대선 불복하겠다는 건가  2022-11-21
929120               [社說] '韓 전술핵' 논의하는 것 자체가 북·중에 메시지 될 것  2022-11-21
929121                [社說] '복심' 정진상 구속, 李대표 턱밑까지 온 대장동 수사  2022-11-21
929122                      [전면광고] 세상을 바꾸는 K-BAB - KB금융그룹  2022-11-21

[929123 rows x 2 columns]

In [142]:
data = data.rename(columns={'0':'title'})
data.head(5)

title        date
0                      경제전문가·일반인 긴급 여론조사;“구조조정만이 살길”  2001-01-01
1                            ◇희망의 2001년… 다시 한번 뛰어보자;  2001-01-01
2                       클린턴이 임명한 테닛;부시, CIA국장 유임시킬듯;  2001-01-01
3  민주의원 이적 새해정국 파문;3명 자민련 입당… 교섭단체 가능 야 강력반발… 영수회...  2001-01-01
4              공적자금 투입 4개은 금융노조 전격 탈퇴;한빛·서울도 구조조정 동의  2001-01-01

In [143]:
data[data.duplicated() == True] # 5292개 중복 타이틀 확인 -> 선별적 삭제 필요
# data[data.title.duplicated() == True] # 63490개 중복 확인 -> 선별적 삭제 필요

title        date
71                                                 시사-교양;  2001-01-01
79                                                    인사;  2001-01-01
102                                  신춘문예 당선작;단편소설;봄 김주옥;  2001-01-01
667                                                문화게시판;  2001-01-05
689                                                문화게시판;  2001-01-05
...                                                   ...         ...
926130             [통판광고] 한국산업의 자부심 - 2022 한국산업의 고객만족도 1위  2022-10-17
926736             [통판광고] 금융도 내 본능대로 SOL Shinhan|쏠 - 신한은행  2022-10-24
927415                       [통판광고] 금융도 내 본능대로 SOL - 신한은행  2022-11-01
928672             [통판광고] 초대형 Neo QLED 8K로 선수교체 - SAMSUNG  2022-11-16
928773  [통판광고] 삼성 신혼가전과 함께 가장 우리다운 집들-이 많아지고 있습니다 - SA...  2022-11-17

[5292 rows x 2 columns]

In [144]:
data = data.drop_duplicates(['title']) # by default: keep = 'first' 
data

title        date
0                           경제전문가·일반인 긴급 여론조사;“구조조정만이 살길”  2001-01-01
1                                 ◇희망의 2001년… 다시 한번 뛰어보자;  2001-01-01
2                            클린턴이 임명한 테닛;부시, CIA국장 유임시킬듯;  2001-01-01
3       민주의원 이적 새해정국 파문;3명 자민련 입당… 교섭단체 가능 야 강력반발… 영수회...  2001-01-01
4                   공적자금 투입 4개은 금융노조 전격 탈퇴;한빛·서울도 구조조정 동의  2001-01-01
...                                                   ...         ...
929117                          [태평로] 손흥민의 웃음을 오랫동안 보고 싶다  2022-11-21
929118  [강헌의 히스토리 인 팝스] (138) Michelle Obama 'This is ...  2022-11-21
929119               [社說] 공공연히 ＂윤석열 퇴진＂ 민주당, 대선 불복하겠다는 건가  2022-11-21
929120               [社說] '韓 전술핵' 논의하는 것 자체가 북·중에 메시지 될 것  2022-11-21
929121                [社說] '복심' 정진상 구속, 李대표 턱밑까지 온 대장동 수사  2022-11-21

[865633 rows x 2 columns]

# 중복값 제거 - 전체 (괄호 안은 갯수)
1) 부분 일치(혹은 포함)
- 통판광고 (295302)
- 전면광고 (125062)
- 신춘문예 (275)
- 당선작 (185)
- 일사일언 (2958)
- 동정 (309)
- 생활한자 (799)
- 리빙 포인트 (565)
- 리빙포인트 (4765)
- hot & cold (3)
- 기업공시 (9)
- 부음 (4456)
- 플라자 (2488)
- 편집자에게 (1841)
- 신문은 선생님 (8069)
- 알림 (53)
- 사진 (29578) < 이대로 빼는게 맞을까?
- 시사-교양 (32)
- 시사교양 (10)
- 확성기 (120)
- 기고 (3877) < 이대로 빼는게 맞을까?
- 의견 (1)
- 이규태 (1380)
- 경제 브리핑 
- 날씨 이야기
- 문화게시판 (102)
- 쇼트트랙
- 취재메모
- 아이스하키
- 만물상
- 태펴올
- 시네마
- 문화나들이 어디로 할까 (49)
- 방송가
- 튼튼스타
- 단신
- 새의자
- 시시각각
- 스포츠
- TEPS
- 연극리뷰
- 클래식
- 부부의 속마음
- 병원 Q & A
- 여행수첩
- 여행단신
- 그래픽 뉴스
- 오태진
- 내정
- 사진기사
- 이곳이 좋아요
- 알뜰정보
- 학술 소식
- 조용헌
- 알립니다
- 호남사람들

- [도서] 
- [강원] [충청] 등 [지역명] +a

2) ~~완벽 일치~~
- 시사 ()
- 교양 ()
- 인사 () < 이대로 빼는게 맞을까?
- 시네마 (65)
- 시시각각 (4)
- 단신; (54)
- 새의자; (3)

In [145]:
stoplist = ['통판광고','전면광고','신춘문예', '당선작', '일사일언', '동정', '생활한자', '리빙 포인트', '리빙포인트', 'hot & cold', '기업공시', '부음','플라자','편집자에게','신문은 선생님','알림','사진', '시사', '교양', '확성기', '기고', '의견', '이규태', '경제 브리핑', '날씨 이야기', '문화게시판', '쇼트트랙', '취재메모', '아이스하키', '만물상', '태평로', '시네마', '문화나들이 어디로 할까', '방송가', '튼튼스타', '단신', '새의자', '시시각각', '스포츠', 'TEPS', '연극리뷰', '클래식', '부부의 속마음', '병원 Q & A', '여행수첩', '여행단신', '그래픽 뉴스', '오태진', '내정', '사진기사', '이곳이 좋아요', '알뜰정보', '학술 소식', '조용헌', '알립니다', '호남사람들', '인사;']

In [146]:
df_result = pd.DataFrame()

for i in stoplist:
    result = data[data['title'].str.contains(i)]
    df_result = pd.concat([df_result,result])
    print(df_result)

                                                    title        date
138135                                     [통판광고] 동작성원쌍떼빌  2003-04-22
138421                                    [통판광고] 동작 성원쌍떼빌  2003-04-24
138511                                      [통판광고] UTOPIA  2003-04-25
138673                                        [통판광고] 하이마트  2003-04-26
138838                                     [통판광고] 돈암동 오페스  2003-04-28
...                                                   ...         ...
926129             [통판광고] 한국산업의 자부심 - 2022 한국산업의 고객만족도 1위  2022-10-17
926735             [통판광고] 금융도 내 본능대로 SOL Shinhan|쏠 - 신한은행  2022-10-24
927414                       [통판광고] 금융도 내 본능대로 SOL - 신한은행  2022-11-01
928671             [통판광고] 초대형 Neo QLED 8K로 선수교체 - SAMSUNG  2022-11-16
928772  [통판광고] 삼성 신혼가전과 함께 가장 우리다운 집들-이 많아지고 있습니다 - SA...  2022-11-17

[1777 rows x 2 columns]
                                         title        date
138135                          [통판광고] 동작성원쌍떼빌  2003-04-22
138421           

In [147]:
# 확인용! (위의 137091 - 134869 = 2222)
df_test = df_result.drop_duplicates('title')
df_test.shape

(134869, 2)

In [148]:
data = pd.merge(data, df_result, how='outer', indicator=True)
data

title        date  \
0                           경제전문가·일반인 긴급 여론조사;“구조조정만이 살길”  2001-01-01   
1                                 ◇희망의 2001년… 다시 한번 뛰어보자;  2001-01-01   
2                            클린턴이 임명한 테닛;부시, CIA국장 유임시킬듯;  2001-01-01   
3       민주의원 이적 새해정국 파문;3명 자민련 입당… 교섭단체 가능 야 강력반발… 영수회...  2001-01-01   
4                   공적자금 투입 4개은 금융노조 전격 탈퇴;한빛·서울도 구조조정 동의  2001-01-01   
...                                                   ...         ...   
867850                          [태평로] 손흥민의 웃음을 오랫동안 보고 싶다  2022-11-21   
867851  [강헌의 히스토리 인 팝스] (138) Michelle Obama 'This is ...  2022-11-21   
867852               [社說] 공공연히 ＂윤석열 퇴진＂ 민주당, 대선 불복하겠다는 건가  2022-11-21   
867853               [社說] '韓 전술핵' 논의하는 것 자체가 북·중에 메시지 될 것  2022-11-21   
867854                [社說] '복심' 정진상 구속, 李대표 턱밑까지 온 대장동 수사  2022-11-21   

           _merge  
0       left_only  
1       left_only  
2       left_only  
3       left_only  
4       left_only  
...           ...  
867850       both  
867851  left_only  
867852  left_only  
867853  left_only  
867854  left_only  

[867855 rows x 3 columns]

In [149]:
data = data.query('_merge == "left_only"').drop(columns=['_merge'])
data

# 865633 - 136958 = 728675 여야 함 (약 2,215개 잉여: 최초 data에서 stoplist 해당 단어 포함 타이틀 존재)

title        date
0                           경제전문가·일반인 긴급 여론조사;“구조조정만이 살길”  2001-01-01
1                                 ◇희망의 2001년… 다시 한번 뛰어보자;  2001-01-01
2                            클린턴이 임명한 테닛;부시, CIA국장 유임시킬듯;  2001-01-01
3       민주의원 이적 새해정국 파문;3명 자민련 입당… 교섭단체 가능 야 강력반발… 영수회...  2001-01-01
4                   공적자금 투입 4개은 금융노조 전격 탈퇴;한빛·서울도 구조조정 동의  2001-01-01
...                                                   ...         ...
867849                           [특파원 리포트] 中 '제로 코로나' 딜레마  2022-11-21
867851  [강헌의 히스토리 인 팝스] (138) Michelle Obama 'This is ...  2022-11-21
867852               [社說] 공공연히 ＂윤석열 퇴진＂ 민주당, 대선 불복하겠다는 건가  2022-11-21
867853               [社說] '韓 전술핵' 논의하는 것 자체가 북·중에 메시지 될 것  2022-11-21
867854                [社說] '복심' 정진상 구속, 李대표 턱밑까지 온 대장동 수사  2022-11-21

[730764 rows x 2 columns]

In [150]:
data['title'] = [re.sub('[^ \가-힣]', '', s) for s in data['title']]
data

# 최종 가용데이터 : 730764 개

title        date
0                               경제전문가일반인 긴급 여론조사구조조정만이 살길  2001-01-01
1                                        희망의 년 다시 한번 뛰어보자  2001-01-01
2                                  클린턴이 임명한 테닛부시 국장 유임시킬듯  2001-01-01
3       민주의원 이적 새해정국 파문명 자민련 입당 교섭단체 가능 야 강력반발 영수회담 거부 검토  2001-01-01
4                      공적자금 투입 개은 금융노조 전격 탈퇴한빛서울도 구조조정 동의  2001-01-01
...                                                   ...         ...
867849                                특파원 리포트  제로 코로나 딜레마  2022-11-21
867851                              강헌의 히스토리 인 팝스          2022-11-21
867852                       공공연히 윤석열 퇴진 민주당 대선 불복하겠다는 건가  2022-11-21
867853                         전술핵 논의하는 것 자체가 북중에 메시지 될 것  2022-11-21
867854                         복심 정진상 구속 대표 턱밑까지 온 대장동 수사  2022-11-21

[730764 rows x 2 columns]

# EDA 후 조선일보 데이터 별도 CSV 파일 저장

In [152]:
data.to_csv(path_or_buf=r"C:\Users\maeve\AIB 14\Section6\Codestates Project 2\chosun_clean.csv", encoding = 'utf-8')

# 번외. 데이터프레임 전체 행/열 보기 설정

In [106]:
pd.options.display.max_rows = 1000
# pd.set_option('display.max_columns', None) # 모든 열을 출력

In [107]:
data

title        date
0                           경제전문가·일반인 긴급 여론조사;“구조조정만이 살길”  2001-01-01
1                                 ◇희망의 2001년… 다시 한번 뛰어보자;  2001-01-01
2                            클린턴이 임명한 테닛;부시, CIA국장 유임시킬듯;  2001-01-01
3       민주의원 이적 새해정국 파문;3명 자민련 입당… 교섭단체 가능 야 강력반발… 영수회...  2001-01-01
4                   공적자금 투입 4개은 금융노조 전격 탈퇴;한빛·서울도 구조조정 동의  2001-01-01
...                                                   ...         ...
867842                           [특파원 리포트] 中 '제로 코로나' 딜레마  2022-11-21
867844  [강헌의 히스토리 인 팝스] (138) Michelle Obama 'This is ...  2022-11-21
867845               [社說] 공공연히 ＂윤석열 퇴진＂ 민주당, 대선 불복하겠다는 건가  2022-11-21
867846               [社說] '韓 전술핵' 논의하는 것 자체가 북·중에 메시지 될 것  2022-11-21
867847                [社說] '복심' 정진상 구속, 李대표 턱밑까지 온 대장동 수사  2022-11-21

[730763 rows x 2 columns]